python/cudf/cudf/utils/ioutils.py

```diff
@@ -473,8 +473,36 @@
 engine : {{ 'auto', 'cudf', 'cudf_experimental', 'pandas' }}, default 'auto'
     Parser engine to use. If 'auto' is passed, the engine will be
     automatically selected based on the other parameters.
-orient : string,
-    Indication of expected JSON string format (pandas engine only).
+lines : boolean, default False
+    Read the file as a json object per line.
+dtype : boolean or dict, default True
+    If True, infer dtypes, if a dict of column to dtype, then use those,
+    if False, then don't infer dtypes at all, applies only to the data.
+byte_range : list or tuple, default None (cudf engine only)
```

Comment:
We could accept an iterable in the python level and tuplify it before handing off to cython, like this:
```
diff --git a/python/cudf/cudf/io/json.py b/python/cudf/cudf/io/json.py
index 2a0ae56597..5133ed1bbc 100644
--- a/python/cudf/cudf/io/json.py
+++ b/python/cudf/cudf/io/json.py
@@ -76,7 +76,7 @@ def read_json(
             dtype,
             lines,
             compression,
-            byte_range,
+            tuple(byte_range) if byte_range else None,
             engine == "cudf_experimental",
             keep_quotes,
         )
```

And then we could write:

```
byte_range : optional iterable, default None (cudf or cudf_experimental engines only)
```